tss-esapi/src/structures/buffers.rs

```diff
@@ -199,7 +201,68 @@ pub mod nonce {
 }
 
 pub mod public_key_rsa {
-    buffer_type!(PublicKeyRSA, 512, TPM2B_PUBLIC_KEY_RSA);
+    use crate::{interface_types::key_bits::RsaKeyBits, tss2_esys::TPM2_MAX_RSA_KEY_BYTES};
+    use std::convert::TryInto;
+    buffer_type!(
+        PublicKeyRsa,
+        TPM2_MAX_RSA_KEY_BYTES as usize,
+        TPM2B_PUBLIC_KEY_RSA
+    );
+
+    impl PublicKeyRsa {
+        pub fn new_empty_with_size(rsa_key_bits: RsaKeyBits) -> Self {
+            match rsa_key_bits {
+                RsaKeyBits::Rsa1024 => PublicKeyRsa(vec![0u8; 128].into()),
+                RsaKeyBits::Rsa2048 => PublicKeyRsa(vec![0u8; 256].into()),
+                RsaKeyBits::Rsa3072 => PublicKeyRsa(vec![0u8; 384].into()),
+                RsaKeyBits::Rsa4096 => PublicKeyRsa(vec![0u8; 512].into()),
+            }
+        }
+    }
+
+    impl TryFrom<PublicKeyRsa> for [u8; 128] {
+        type Error = Error;
+
+        fn try_from(value: PublicKeyRsa) -> Result<Self> {
+            value
+                .value()
+                .try_into()
+                .map_err(|_| Error::local_error(WrapperErrorKind::WrongParamSize))
+        }
+    }
```

Comment:
Similarly, these fail compilation because `TryFrom<&[u8]> for [u8; ...]` (at least for this size of the array) is quite recent, but for this one I can just disable compilation for older versions.

Reply:
I will fix this by rewriting the try_from functions for those array sizes.

Reply:
I have rewritten this now so it compiles with 1.43.0